lib/simplifyExpression/collectAndCombineSearch/addLikeTerms.js

```diff
@@ -140,28 +162,28 @@ function addNegativeOneCoefficient(node) {
   }
 }
 
-// Given a sum of like polynomial terms, groups the coefficients
+// Given a sum of like terms, groups the coefficients
 // e.g. 2x^2 + 3x^2 + 5x^2 -> (2+3+5)x^2
 // Returns a Node.Status object.
-function groupCoefficientsForAdding(node) {
+function groupCoefficientsForAdding(node, termWithCoefficientSubclass) {
   let newNode = clone(node);
 
-  const polynomialTermList = newNode.args.map(n => new Node.PolynomialTerm(n));
-  const coefficientList = polynomialTermList.map(p => p.getCoeffNode(true));
+  const termList = newNode.args.map(n => new termWithCoefficientSubclass(n));
+  const coefficientList = termList.map(p => p.getCoeffNode(true));
```

Comment:
since you're naming them terms here, i think you can just do term (and not term with coefficient) as the name - I think that'd be fine

also here `p => p.getCoeffNode(true));` --> `term => term.getCoeffNode(true));`